src/rl/rl_controller/rl_controller/controller.py

```diff
@@ -0,0 +1,222 @@
+import rclpy
+from rclpy.node import Node
+from std_msgs.msg import Bool, String, Float64
+from geometry_msgs.msg import Pose, PoseArray
+import threading
+import time
+import numpy as np
+from gymnasium import Env
+from gymnasium.spaces import Discrete, Dict, Box
+
+# Constants
+CAR_POSITION = Pose()
+SELECTED_TRAJECTORY = None
+DESIRED_SPEED = 3.0
+TRACK_POINT_REACHED = 0
+TURNING_ANGLES = []
+
+class RLEnvironmentNode(Node):
```

Comment:
Please write a little doc string about what this env dose, and please be more specific since when you create more envs that do different things you don't wanna just delete the old ones since its good to compare